test/aida/bot_test.exs

```diff
@@ -94,6 +94,80 @@ defmodule Aida.BotTest do
       %{bot: bot, initial_session: initial_session}
     end
 
+    test "forward text messages", %{
+      bot: bot,
+      initial_session: initial_session
+    } do
+      forward_messages_id = Ecto.UUID.generate()
+      session = initial_session |> Session.put(".forward_messages_id", forward_messages_id)
+
+      with_mock HTTPoison,
+        post: fn _url, _body -> {:ok, %HTTPoison.Response{status_code: 200}} end do
+        %{reply: reply} = Message.new("a message", bot, session) |> Bot.chat()
+
+        assert length(reply) == 0
+
+        assert MessageLog
+               |> Repo.all()
+               |> Enum.filter(&(&1.direction == "outgoing"))
+               |> Enum.count() == 0
+
+        assert MessageLog
+               |> Repo.all()
+               |> Enum.filter(&(&1.direction == "incoming"))
+               |> Enum.count() == 1
+
+        assert called(
+                 HTTPoison.post(
+                   "#{bot.notifications_url}/messages/#{forward_messages_id}",
+                   %{type: "text", direction: "uto", content: "a message"}
+                   |> Poison.encode!()
+                 )
+               )
+      end
+    end
+
+    test "does not forward when session with blank .forward_messages_id", %{
```

Comment:
Maybe we could prevent this case altogether by filtering the forward messages id at the controller level instead of having to filter it on the bot. Either we have a valid `.forward_messages_id` or we don't have anything, no need to specifically test and escape the empty string case.

Reply:
This test was written to support an actual case for the current instedd/aida-ui#231. Maybe we should prevent this case at the controller level, but I don't see any bad side effect on leaving this test (and its corresponding guard in bot) as a safeguard. What do you think?

Reply:
I think that if we start doing this kind of things, in short we will be checking everything in the session and that's far from ideal.

I think it's better to ensure that the session is consistent.

It can be done at the session level. Maybe we can ensure that `Session.put` won't store any invalid value and clear the key instead, the same way we are already doing it with `nil` to avoid keys without a value.